test/e2e/operator_test.go

```diff
@@ -1829,6 +1829,127 @@ func TestLoadBalancingAlgorithmUnsupportedConfigOverride(t *testing.T) {
+		t.Fatalf("failed to create configmap %q: %v", errorPageConfigmapName, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), errorPageConfigmap); err != nil {
+			t.Fatalf("failed to delete configmap %q: %v", errorPageConfigmapName, err)
+		}
+	}()
+
+	conditions := []operatorv1.OperatorCondition{
+		{Type: operatorv1.IngressControllerAvailableConditionType, Status: operatorv1.ConditionTrue},
+		{Type: operatorv1.LoadBalancerManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+		{Type: operatorv1.DNSManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+	}
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, conditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// The controller should create a configmap in "openshift-ingress".
+	cmName := controller.HttpErrorCodePageConfigMapName(ic)
+	cm := &corev1.ConfigMap{}
+	err := wait.PollImmediate(1*time.Second, 5*time.Minute, func() (bool, error) {
+		if err := kclient.Get(context.TODO(), cmName, cm); err != nil {
+			t.Logf("failed to get configmap %q: %v", cmName, err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to observe initial configmap %q: %v", cmName, err)
+	}
+	if actual, expected := cm.Data["error-page-503.http"], errorPageConfigmap.Data["error-page-503.http"]; actual != expected {
+		t.Errorf("failed to observe expected data in configmap %q: expected %q, got %q", cmName, expected, actual)
+	}
+
+	// The deployment should use the custom error-page configmap.
+	deploymentName := controller.RouterDeploymentName(ic)
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), deploymentName, deployment); err != nil {
+		t.Fatalf("failed to get deployment %q: %v", deploymentName, err)
+	}
+	if err := waitForDeploymentEnvVar(t, kclient, deployment, 1*time.Minute, "ERRORFILE_503", "/var/lib/haproxy/conf/error_code_pages/error-page-503.http"); err != nil {
+		t.Fatalf("expected deployment %q to use the custom error-page file: %v", deploymentName, err)
+	}
+
+	// The controller should recreate the configmap if it is deleted.
+	if err := kclient.Delete(context.TODO(), cm); err != nil {
+		t.Fatalf("failed to delete configmap %q: %v", cmName, err)
+	}
+	err = wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		if err := kclient.Get(context.TODO(), cmName, cm); err != nil {
+			t.Logf("failed to get configmap %q: %v", cmName, err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to observe recreated configmap %q: %v", cmName, err)
+	}
+
+	// The controller should update the its configmap if the user's changes.
+	errorPageConfigmap.Data["error-page-503.http"] = `HTTP/1.0 503 Service Unavailable
+Pragma: no-cache
+Cache-Control: private, max-age=0, no-cache, no-store
+Connection: close
+Content-Type: text/text
+
+Not found!
+`
```

Comment:
I think the [force-push from ce3addc to 7f5c69b](https://github.com/openshift/cluster-ingress-operator/compare/ce3addc828a9c2e2218b61837376aa54777917a9..7f5c69bc891d527974bfecabc8eee78cc04f5e61) dropped the linefeeds.  I'm not sure how that happened, and maybe it doesn't _really_ matter, but it might be worth adding a comment that this example is technically invalid.